client/my-sites/domains/domain-management/settings/cards/domain-diagnostics-card.tsx

```diff
@@ -0,0 +1,171 @@
+		return null;
+	}
+
+	const renderDiagnosticForRecord = ( record: string ) => {
+		const uppercaseRecord = record.toUpperCase();
+		const records = emailDnsDiagnostics.records;
+
+		if ( records[ record ].status === 'incorrect' ) {
+			return (
+				<li key={ record }>
+					{ sprintf(
+						/* translators: dnsRecordType is a DNS record type, e.g. SPF, DKIM or DMARC */
+						translate(
+							'Your %(dnsRecordType)s record is incorrect. The correct record should be:'
+						),
+						{
+							dnsRecordType: uppercaseRecord,
+						}
+					) }
+					<p>
+						<code>{ records[ record ].correct_record }</code>
+					</p>
+				</li>
+			);
+		} else if ( records[ record ].status === 'not_found' ) {
+			return (
+				<li key={ record }>
+					{ sprintf(
+						/* translators: dnsRecordType is a DNS record type, e.g. SPF, DKIM or DMARC */
+						translate( 'There is no %(dnsRecordType)s record. The correct record should be:' ),
+						{
+							dnsRecordType: uppercaseRecord,
+						}
+					) }
+					<p>
+						<code>{ records[ record ].correct_record }</code>
+					</p>
+				</li>
+			);
+		}
+
+		return null;
+	};
+
+	const renderFixInstructions = () => {
+		return (
+			<p>
+				{ translate(
+					"To fix these issues, you should go to your domain's DNS provider and add the records above to your domain's DNS settings."
+				) }
+			</p>
+		);
+	};
+
+	const fixDnsIssues = () => {
+		setIsRestoringDefaultRecords( true );
+
+		wpcom.req
+			.post( {
+				apiNamespace: 'wpcom/v2',
+				path: `/domains/dns/restore-default-email-records/${ domain.name }`,
+			} )
+			.then( () => {
+				dispatch(
+					successNotice(
+						translate( 'The default email DNS records were successfully fixed!' ),
+						noticeOptions
+					)
+				);
+				// Fetch DNS records to update the DNS records section's UI
+				dispatch( fetchDns( domain.name, true ) );
+				refetchDomainDiagnostics();
+			} )
+			.catch( () => {
+				dispatch(
+					errorNotice(
+						translate( 'There was a problem when restoring default email DNS records' ),
+						noticeOptions
+					)
+				);
+				return;
+			} )
+			.finally( () => {
+				setIsRestoringDefaultRecords( false );
+			} );
+	};
+
+	const renderFixButton = () => {
+		return (
+			<Button
+				busy={ isRestoringDefaultRecords }
+				disabled={ isRestoringDefaultRecords }
+				onClick={ fixDnsIssues }
+				primary
+			>
+				{ translate( 'Fix DNS issues automatically' ) }
+			</Button>
+		);
+	};
+
+	return (
+		<Accordion
+			className="domain-diagnostics-card__accordion"
+			title={ translate( 'Diagnostics', { textOnly: true } ) }
+			subtitle={ translate( 'There are some issues with your domain', { textOnly: true } ) }
+			key="diagnostics"
+			expanded
+		>
+			<div>
+				<Notice
+					status="is-warning"
+					text={ translate( "There are some issues with your domain's email DNS settings." ) }
+					showDismiss={ false }
+				/>
+				<ul>{ recordsToCheck.map( ( record ) => renderDiagnosticForRecord( record ) ) }</ul>
```

Comment:
You can simplify the `.map` callback:
```suggestion
				<ul>{ recordsToCheck.map( renderDiagnosticForRecord ) }</ul>
```

Reply:
Nice!